File: Qubit Server/dev/src/main/java/org/labrad/qubits/config/MicrowaveSourceConfig.java
package org.labrad.qubits.config;

import org.labrad.qubits.resources.MicrowaveSource;

public interface MicrowaveSourceConfig {
  public boolean isOn();
  public double getFrequency();
  public double getPower();

  public SetupPacket getSetupPacket(MicrowaveSource src);
}

File: QubitServer/src/main/scala/org/labrad/qubits/config/MicrowaveSourceConfig.scala
package org.labrad.qubits.config

import org.labrad.qubits.resources.MicrowaveSource

trait MicrowaveSourceConfig {
  def isOn: Boolean
  def frequency: Double
  def power: Double

  def getSetupPacket(src: MicrowaveSource): SetupPacket
}
